File: livebox/src/test/java/com/creations/livebox/util/FakeSource.java
package com.creations.livebox.util;

import com.creations.livebox.datasources.LocalDataSource;

import java.lang.reflect.Type;

/***
 * Fake source used on testing
 * @author Sérgio Serra
 */
public class FakeSource<I, T> implements LocalDataSource<I, T> {

    public static <I, T> FakeSource<I, T> create() {
        return new FakeSource<>();
    }

    private I data;

    @Override
    public Optional<T> read(String key) {
        //noinspection unchecked
        return Optional.ofNullable((T) data);
    }

    @Override
    public void save(String key, I input) {
        data = input;
    }

    @Override
    public void clear(String key) {

    }

    @Override
    public Type getType() {
        return null;
    }
}

File: livebox/src/test/java/com/creations/livebox/util/FakeSource.kt
package com.creations.livebox.util

import com.creations.livebox.datasources.LocalDataSource

import java.lang.reflect.Type

/***
 * Fake source used on testing.
 * @author Sérgio Serra
 */
open class FakeSource<I, T> : LocalDataSource<I, T> {

    private var data: I? = null

    override val type: Type = Any::class.java

    override fun read(key: String): T? {
        @Suppress("UNCHECKED_CAST")
        return data as T?
    }

    override fun save(key: String, input: I) {
        data = input
    }

    override fun clear(key: String) {}

    companion object {
        @JvmStatic
        fun <I, T> create(): FakeSource<I, T> = FakeSource()
    }
}
